src/Marten/Events/AsyncProjectionTestingExtensions.cs

```diff
@@ -77,4 +94,8 @@ public static async Task WaitForNonStaleProjectionDataAsync(this IMartenDatabase
                 $"The projections timed out before reaching the initial sequence of {initial.EventSequenceNumber}");
         }
     }
+
+    private static int AsyncProjectionsCount(this IDocumentStore store)
+        => store.Options.Events.Projections().Count(p => p.Lifecycle == ProjectionLifecycle.Async)
+           + new[] { ShardState.HighWaterMark }.Length;
```

Comment:
not 100% sure if it is the best way to get **all** AsyncProjectionCount.. 
at least, i tried to explicitly reference `ShardState.HighWaterMark` and not just add `+1`